cookbooks/sidekiq6/recipes/setup.rb

```diff
@@ -0,0 +1,95 @@
+#
+# Cookbook Name:: sidekiq
+# Recipe:: setup
+#
+
+if node['sidekiq']['is_sidekiq_instance']
+  # report to dashboard
+  ey_cloud_report "sidekiq6" do
+    message "Setting up sidekiq 6 or later"
+  end
+
+  # loop through applications
+  node['dna']['applications'].each do |app_name, _|
+    node['sidekiq']['workers'].times do |count|
+      execute "restart-sidekiq-for-#{app_name}" do
+        command "systemctl daemon-reload"#&& systemctl restart sidekiq_#{app_name}_#{count}"
+        action :nothing
+      end
```

Comment:
This is not working (independently of un-commenting the restart part). When there is more than 1 worker, the configs are properly installed but the daemons of each worker are not restarted.  In order to make it work when there is more than 1 worker (with the yml files defined in the custom recipe) I needed to add the count in the name of the execution, like that:
```suggestion
      execute "restart-sidekiq-for-#{app_name}-#{count}" do
        command "systemctl daemon-reload && systemctl restart sidekiq_#{app_name}_#{count}"
        action :nothing
      end
```
By doing that, of course, it has to be changed where it is called everywhere passing the `count`. It is doable in every place without issue as we always have the `count` except in 1 spot with the definition in the database.yml.  In our project, we don't use it so I just did a quick fix by keeping the same definition as you did but outside of the loop, just in order to not have an issue with the recipe.

Reply:
This looks good to me. I'll have the team test it out. I think I was originally thinking of something similar looking at the commented out part.